transcom/mymove
# INTEGRATION-B-18434 - (part 2) Fix issue related to PPM update and MultiMove changes in INT

## [B-18434](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A883558)
### [E-05255](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic%3A908778)

## Summary

**For INTEGRATION**

NOTE: This a second PR for INT.  It contains the following:
- Address BAD MERGE(accidentally merged integrationTesting) into B-18434(main). Had to rebase prior to bad merge and cherry picked change sets.
- Fix for PPM update where secondary addresses were accidentally deleted on an update because system erroneously assumed NULL values for secondary addresses implied deletions. Adding two new optional flags to explicitly tell system to delete.
- Fix current issue in INTEGRATION where PPM secondary addresses are not displaying in edit view. This is because recent Multimove features changed the way the view retrieved move information via new API call. This new API was not returning back ALL PPM addresses from the service layer.
- Fix issue on PPM edit view where previously saved CloseOutOffice was not properly displaying on the Closeout Office Search widget. This was because the service layer was not explicitly loading and returning the CloseoutOffice.Address object. It was originally returning NULL. This is related to recent Multimove changes.

This PR will encompass the entire workflow for the zip to address enhancement for PPM onboarding for the customer side(INTERNAL). It will also include [B-17919](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A868302) because B-17919 cannot exist alone without the backend feature.

Address will replace ZIP code. The following have been added to ppm_shipments table:

- pickupAddressID (required)
- secondaryPickupAddressID
- destinationAddressID (required)
- secondaryDestinationAddressID
- hasSecondaryPickupAddress
- hasSecondaryDestinationAddress

ID is a foreignKey to Address table. 

The customer PPM onboarding process will remain the same. Only difference will be instead of "Use ... zip" it will be "Use my ... address"
![Screenshot 2024-02-22 at 12 15 37 PM](https://github.com/transcom/mymove/assets/148127004/bed3425c-a614-436a-9bdd-e79a72cbd14c)

![Screenshot 2024-02-22 at 12 17 04 PM](https://github.com/transcom/mymove/assets/148127004/204603a2-52c1-4223-9bf0-c45f9fbcc5d6)

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the
2. Login as a customer
3. create PPM and apply all test case permutations
- Use current origin address
- Use my current destination address
- Do not use current origin address
- Do not use my current destination address
- Apply will you add items to your PPM from a different address?
- Apply Will you deliver part of your PPM to a different address?
4. update PPM and apply all test case permutations
- Use current origin address
- Use my current destination address
- Do not use current origin address
- Do not use my current destination address
- Apply will you add items to your PPM from a different address?
- Apply Will you deliver part of your PPM to a different address?

NOTE: MAKE SURE TO TEST BOTH CREATION AND UPDATE OF PPMS!!!!!!!!! 

### Frontend


### Backend
